pebblo/app/config/models.py

```diff
@@ -0,0 +1,235 @@
+import importlib.util
+import logging
+import os
+from typing import Optional, Union
+
+from pydantic import Field, field_validator, model_validator
+from pydantic_settings import BaseSettings
+
+from pebblo.app.config import utils
+from pebblo.app.config.utils import (
+    DEFAULT_LOG_BACKUP_COUNT,
+    DEFAULT_LOG_FILE,
+    DEFAULT_LOG_LEVEL,
+    DEFAULT_LOG_MAX_FILE_SIZE,
+    dir_path,
+    expand_path,
+    update_anonymize_snippets_exists,
+)
+from pebblo.app.enums.common import (
+    ClassificationMode,
+    DBStorageTypes,
+    ReportFormat,
+    ReportLibraries,
+    StorageTypes,
+)
+
+
+# Port BaseModel
+class DaemonConfig(BaseSettings):
+    host: str = Field(default="localhost")
+    port: int = Field(default=8000)
+
+    @field_validator("host")
+    @classmethod
+    def host_should_be_str(cls, host: str) -> str:
+        if not isinstance(host, str):
+            raise ValueError(f"Error: Invalid host '{host}'. Host must be a string.")
+        return host
+
+    @field_validator("port")
+    @classmethod
+    def check_port_validity(cls, port: str) -> int:
+        try:
+            port = int(port)
+        except ValueError:
+            raise ValueError(
+                f"Error: Invalid port value '{port}'. Port must be an integer."
+            )
+        if not (0 < port <= 65535):
```

Comment:
When invalid value like `abc` is passed to port, this statement will give error. This check need to be inside try-except block above.

Reply:
If we are passing port as `abc` it is coming with proper error:

```
Error while loading config details, err: 1 validation error for Config
daemon.port
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='abc', input_type=str]
```

Reply:
Removed custom validation for int type checking, as pydantic is already doing it while conversion.